src/lib/y2partitioner/dialogs/main.rb

```diff
@@ -67,28 +69,28 @@ def next_button
         Yast::Mode.installation ? Yast::Label.AcceptButton : Yast::Label.FinishButton
       end
 
-      # runs dialog.
-      # @param system [Y2Storage::DeviceGraph] system graph used to detect if something
-      # is going to be formatted
-      # @param initial [Y2Storage::DeviceGraph] device graph to display
-      # @return [Symbol] result of dialog
+      # Runs the dialog
+      #
+      # @param system [Y2Storage::Devicegraph] system graph used to detect if something
+      #   is going to be formatted.
+      # @param initial [Y2Storage::Devicegraph] device graph to display.
+      #
+      # @return [Symbol] result of the dialog.
       def run(system, initial)
-        DeviceGraphs.create_instance(system, initial)
         UIState.create_instance
-        res = nil
-        loop do
-          res = super()
+        DeviceGraphs.create_instance(system, initial)
+
+        return :back unless accepted_partitioner_warning?
```

Comment:
The name is rather confusing. In most cases the warning will not be displayed (either no need to accept it). Calling something called `accepted_partitioner_warning?` and getting `true` in those cases... is not nice for the reader.